lib/elixir/lib/module.ex

```diff
@@ -571,22 +579,27 @@ defmodule Module do
       end
 
   """
-  def add_doc(module, line, kind, tuple, signature \\ [], doc)
+  @spec add_doc(module, non_neg_integer, def_kind | type_kind, function_arity, list,
+                String.t | boolean | nil)
+      :: :ok | {:error, :private_doc}
+  def add_doc(module, line, kind, function_tuple, signature \\ [], doc)
 
-  def add_doc(_module, _line, kind, _tuple, _signature, doc) when kind in [:defp, :defmacrop, :typep] do
+  def add_doc(_module, _line, kind, _function_tuple, _signature, doc)
+      when kind in [:defp, :defmacrop, :typep] do
     if doc, do: {:error, :private_doc}, else: :ok
   end
 
-  def add_doc(module, line, kind, tuple, signature, doc) when
-      kind in [:def, :defmacro, :type, :opaque] and (is_binary(doc) or is_boolean(doc) or doc == nil) do
+  def add_doc(module, line, kind, function_tuple, signature, doc)
+      when kind in [:def, :defmacro, :type, :opaque]
+      and (is_binary(doc) or is_boolean(doc) or doc == nil) do
```

Comment:
And should be in the previous line and the parens should be aligned with the beginning of `kind`.